tests/testthat/test-ps_weighting.R

```diff
@@ -61,3 +61,39 @@ test_that("Imputation works as expected", {
   expect_equal(mean(output_2$propensity_weight), 1)
 
 })
+
+test_that("Imputation works when 1 IV", {
+
+  iv_age <- subset_dat_2 %>%
+    add_propensity_weights(
+      treatment = "treat",
+      ivs = "age",
+      impute_missing = TRUE
+    )
+
+  expect_equal(
+    iv_age %>% dplyr::select(-propensity_weight),
+    subset_dat_2
+  )
+
+  expect_equal(ncol(iv_age), (ncol(subset_dat_2) + 1))
+  expect_true("propensity_weight" %in% colnames(iv_age))
+  expect_equal(mean(iv_age$propensity_weight), 1)
+
+  iv_happiness <- subset_dat_2 %>%
```

Comment:
Is it worth breaking this into a separate test: "Imputation works with one categorical IV"

Reply:
I don't have a preference either way. If you think that makes more sense, I can make the update

Reply:
To me they are separate tests, since they are covering different chunks of code in the function based on the conditional logic you implemented

Reply:
Makes sense, updated